crates/core_simd/src/vector.rs

```diff
@@ -311,6 +311,42 @@ where
         unsafe { self.store(slice.as_mut_ptr().cast()) }
     }
 
+    #[must_use]
+    #[inline]
+    pub fn masked_load_or(slice: &[T], or: Self) -> Self {
+        Self::masked_load_select(slice, Mask::splat(true), or)
+    }
+
+    #[must_use]
+    #[inline]
+    pub fn masked_load_select(slice: &[T], enable: Mask<isize, N>, or: Self) -> Self {
```

Comment:
`isize` here is wrong I think. I got it from `gather` but it made sense there as it was masking pointers. I think we want `Mask<<T as SimdElement>::Mask, N>`

Reply:
I agree--the mask should either match the element type, or possibly accept any element.